crates/miden-objects/src/account/component/mod.rs

```diff
@@ -112,6 +114,59 @@ impl AccountComponent {
             .with_supported_types(template.metadata().supported_types().clone()))
     }
 
+    /// Creates an [`AccountComponent`] from a [`Package`].
+    ///
+    /// This method extracts the account component metadata from the package and creates
+    /// a component with the provided storage slots. The component will have default
+    /// supported types (RegularAccountImmutableCode) unless overridden.
+    ///
+    /// # Arguments
+    ///
+    /// * `package` - The package containing the account component metadata and library
+    /// * `storage_slots` - The storage slots to initialize the component with
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - The package does not contain account component metadata
+    /// - The metadata cannot be deserialized
+    /// - The component creation fails (e.g., too many storage slots)
+    ///
+    /// # Panics
+    ///
+    /// Panics if the package does not contain account component metadata bytes.
+    pub fn from_package(
+        package: &Package,
+        storage_slots: Vec<StorageSlot>,
+    ) -> Result<Self, AccountError> {
+        let metadata_bytes = package
+            .account_component_metadata_bytes
+            .as_deref()
+            .ok_or(AccountError::AccountComponentMetadataMissing)?;
+
+        let metadata =
+            AccountComponentMetadata::read_from_bytes(metadata_bytes).map_err(|err| {
+                AccountError::other_with_source(
+                    "Failed to deserialize account component metadata",
+                    err,
+                )
+            })?;
+
+        let template =
+            AccountComponentTemplate::new(metadata, package.unwrap_library().as_ref().clone());
+
+        let component = AccountComponent::new(template.library().clone(), storage_slots)?;
+
+        // Use supported types from template metadata, or default to RegularAccountUpdatableCode
+        let supported_types = if template.metadata().supported_types().is_empty() {
+            BTreeSet::from_iter([AccountType::RegularAccountImmutableCode])
```

Comment:
There is a mismatch between the comment and the code. I'd probably use both `RegularAccountUpdatableCode` and `RegularAccountImmutableCode` as defaults, so it would generally support any kind of "regular account".